src/views/FarmAuction/components/AuctionDetailsCard/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React from 'react'
+import styled from 'styled-components'
+import {
+  Text,
+  Heading,
+  Card,
+  CardHeader,
+  CardBody,
+  Flex,
+  Spinner,
+  Skeleton,
+  Tag,
+  Button,
+  CheckmarkCircleIcon,
+  useModal,
+} from '@pancakeswap/uikit'
+import { useTranslation } from 'contexts/Localization'
+import { Auction, AuctionStatus, ConnectedBidder } from 'config/constants/types'
+import { getBalanceNumber } from 'utils/formatBalance'
+import PlaceBidModal from '../PlaceBidModal'
+import { AuctionSchedule } from './AuctionSchedule'
+import CannotBidMessage from './CannotBidMessage'
+import AuctionFooter from './AuctionFooter'
+
+const AuctionDetailsCard = styled(Card)`
+  flex: 1;
+`
+
+interface AuctionDetailsProps {
+  auction: Auction
+  conncetedBidder: ConnectedBidder
+  refreshBidders: () => void
+}
+
+const AuctionDetails: React.FC<AuctionDetailsProps> = ({ auction, conncetedBidder, refreshBidders }) => {
+  const { t } = useTranslation()
+
+  const [onPresentPlaceBid] = useModal(
+    <PlaceBidModal
+      conncetedBidder={conncetedBidder}
+      refreshBidders={refreshBidders}
+      initialBidAmount={auction?.initialBidAmount}
+    />,
+  )
+
+  if (!auction) {
+    return (
+      <AuctionDetailsCard mb={['24px', null, null, '0']}>
+        <CardHeader>
+          <Heading>{t('Current Auction')}</Heading>
+        </CardHeader>
+        <CardBody>
+          <Flex justifyContent="center" alignItems="center" height="100%">
+            <Spinner />
+          </Flex>
+        </CardBody>
+      </AuctionDetailsCard>
+    )
+  }
+
+  const notConnectedOrNotWhitelisted = !conncetedBidder || (conncetedBidder && !conncetedBidder.isWhitelisted)
+  const whitelistedAndReadyToBid = !notConnectedOrNotWhitelisted && conncetedBidder.bidderData
+
+  let bidSection = (
```

Comment:
cleaner to make a function and call the function in the render, so you don't have to deal with a let, and and the logic is segregated from the rest

Reply:
fixed in eeb8832817ed3d4433206a271554fa4b84fed980